ivy.el

```diff
@@ -3956,6 +3963,24 @@ characters (previous if ARG is negative)."
   (interactive "p")
   (ivy--yank-by #'forward-char arg))
 
+(defvar ivy--pulse-overlay nil
+  "Overlay used to highlight yanked word.")
+
+(defvar ivy--pulse-timer nil
+  "Timer used to dispose of ivy--pulse-overlays.")
+
+(defun ivy--pulse-region (begin end)
+  (unless ivy--pulse-overlay
+    (setq ivy--pulse-overlay (make-overlay 0 0)))
+  (when ivy--pulse-timer
+    (cancel-timer ivy--pulse-timer))
+  (move-overlay ivy--pulse-overlay begin end (current-buffer))
```

Comment:
Can't this function be simplified a bit? Can we not cancel the timer before initialising the overlay?

```el
(defun ivy--pulse-region (begin end)
  (when ivy--pulse-timer
    (cancel-timer ivy--pulse-timer))
  (if ivy--pulse-overlay
      (move-overlay ivy--pulse-overlay begin end)
    (setq ivy--pulse-overlay (make-overlay begin end)))
  (overlay-put ivy--pulse-overlay 'face 'ivy-yanked-word)
  (setq ivy--pulse-timer (run-at-time 0.5 nil #'ivy--delete-pulse-overlay)))
```

Reply:
> Can we not cancel the timer before initialising the overlay?

what does it change, really?